hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMStateMachine.java

```diff
@@ -179,25 +193,79 @@ public long takeSnapshot() throws IOException {
     long startTime = Time.monotonicNow();
     TermIndex lastTermIndex = getLastAppliedTermIndex();
     long lastAppliedIndex = lastTermIndex.getIndex();
-    TransactionInfo lastAppliedTrxInfo =
-        TransactionInfo.fromTermIndex(lastTermIndex);
-    if (transactionBuffer.getLatestTrxInfo()
-        .compareTo(lastAppliedTrxInfo) < 0) {
+    if (isInitialized.get()) {
+      TransactionInfo lastAppliedTrxInfo =
+          TransactionInfo.fromTermIndex(lastTermIndex);
+      if (transactionBuffer.getLatestTrxInfo().compareTo(lastAppliedTrxInfo)
+          < 0) {
+        transactionBuffer.updateLatestTrxInfo(
+            TransactionInfo.builder().setCurrentTerm(lastTermIndex.getTerm())
+                .setTransactionIndex(lastTermIndex.getIndex()).build());
+        transactionBuffer.setLatestSnapshot(
+            transactionBuffer.getLatestTrxInfo().toSnapshotInfo());
+      } else {
+        lastAppliedIndex =
+            transactionBuffer.getLatestTrxInfo().getTransactionIndex();
+      }
+
+      transactionBuffer.flush();
+      LOG.info("Current Snapshot Index {}, takeSnapshot took {} ms",
+          lastAppliedIndex, Time.monotonicNow() - startTime);
+    }
+    super.takeSnapshot();
```

Comment:
Do we need this call?

Reply:
yes, to ensure if there are any updates to sm directory by the base stateMachine impelementation.